web/services/hosts.go

```diff
@@ -133,6 +135,36 @@ func (s *hostsService) GetAllBySAPSystemID(id string) (models.HostList, error) {
 	return hostList, nil
 }
 
+func (s *hostsService) GetByName(name string) (*models.Host, error) {
+	var host entities.Host
```

Comment:
This is not really needed, the cloud privder logic can be squashed to `GetByID`
`GetByName` could have multiple results